packages/plugin-ext/src/plugin/type-converters.ts

```diff
@@ -29,6 +29,16 @@ import { UriComponents } from '../common/uri-components';
 import { isReadonlyArray } from '../common/arrays';
 import { MarkdownString as MarkdownStringDTO } from '@theia/core/lib/common/markdown-rendering';
 import { isObject } from '@theia/core/lib/common';
+import * as editorRange from '@theia/monaco-editor-core/esm/vs/editor/editor.api';
+import { denamespaceTestTag, ITestErrorMessage, ITestItem, ITestTag, namespaceTestTag, TestMessageType } from '@theia/testing/lib/common/test-types';
+import { TestId } from '@theia/testing/lib/common/test-id';
+import * as htmlContent from '@theia/monaco-editor-core/esm/vs/base/common/htmlContent';
+import { getPrivateApiFor } from './testing-private-api';
+import { parse } from '@theia/monaco-editor-core/esm/vs/base/common/marshalling';
+import { cloneAndChange } from '@theia/monaco-editor-core/esm/vs/base/common/objects';
+import * as languages from '@theia/monaco-editor-core/esm/vs/editor/common/languages';
+import { Location } from '../common/plugin-api-rpc-model';
+import * as marked from '@theia/monaco-editor-core/src/vs/base/common/marked/marked';
```

Comment:
```suggestion
import * as marked from '@theia/monaco-editor-core/esm/vs/base/common/marked/marked';
```

Reply:
Hi @paul-marechal, we have in fact tried importing `marked` with the JavaScript file from `esm` instead of the Typescript from `src` like the other imports but that generates compilation errors (as shown in the screenshot below) suggesting to add the marked.d.ts file in the _marked_ directory where `marked.js` is found following the `esm` path which is a temporary fix as it would result in the same error with a fresh build. We don't know the reason of this and we still didn't find a permanent solution :

![image](https://user-images.githubusercontent.com/57735310/221086995-43774f73-4aed-4668-a1ed-2be6c1d1c903.png)

Reply:
The reason is that the `marked` files in `esm` have no sourcemap because `marked` isn't compiled from source. [`marked` is a library](https://www.npmjs.com/package/marked) that VSCode has in-sourced by adding the compiled JS from a particular commit of `marked`. That means there's no source map in `esm`, so TS can't find its way back to the `d.ts` in the sources. There may be some TS mechanism to import from `esm` (definitely the correct idea) and still point to the `d.ts` in the `src` directory. though.